src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -403,15 +403,15 @@ public TrainTestData TrainTestSplit(IDataView data, double testFraction = 0.1, s
         /// <summary>
         /// Ensures the provided <paramref name="samplingKeyColumn"/> is valid for <see cref="RangeFilter"/>, hashing it if necessary, or creates a new column <paramref name="samplingKeyColumn"/> is null.
         /// </summary>
-        internal static void EnsureGroupPreservationColumn(IHostEnvironment env, ref IDataView data, ref string samplingKeyColumn, uint? seed = null)
+        internal static void EnsureGroupPreservationColumn(IHostEnvironment env, ref IDataView data, ref string samplingKeyColumn, int? seed = null)
         {
             // We need to handle two cases: if samplingKeyColumn is provided, we use hashJoin to
             // build a single hash of it. If it is not, we generate a random number.
 
             if (samplingKeyColumn == null)
             {
                 samplingKeyColumn = data.Schema.GetTempColumnName("SamplingKeyColumn");
-                data = new GenerateNumberTransform(env, data, samplingKeyColumn, seed);
+                data = new GenerateNumberTransform(env, data, samplingKeyColumn, (uint?)seed);
```

Comment:
>GenerateNumberTransform [](start = 27, length = 23)

Seems that the see for `GenerateRandomNumerTransform` is more a random see right? Why not making that a `int`?

Reply:
Probably because the `GenerateNumberTransform` is not part of the public surface area is why he (correctly, I think) judged it as being less important.

---
In reply to: [265333130](https://github.com/dotnet/machinelearning/pull/2942#discussion_r265333130) [](ancestors = 265333130)